File: adventofcode/2021/day02/day22.cpp
#include <bits/stdc++.h>
using namespace std;
/*
part 2
*/
int main() {
    freopen("inputs/input1.txt", "r", stdin);
    freopen("outputs/output2.txt", "w", stdout);
    string input;
    long long depth = 0, hor = 0, aim = 0;
    while (getline(cin, input)) {
        int pos = input.find(" ");
        string direction = input.substr(0, pos);
        int magnitude = stoi(input.substr(pos + 1));
        if (direction == "down") {
            aim += magnitude;
        } else if (direction == "up") {
            aim -= magnitude;
        } else if (direction == "forward") {
            hor += magnitude;
            depth += (magnitude*aim);
        }
    }
    cout<<depth*hor<<endl;
}

File: adventofcode/2021/day02/day22.py
if __name__ == '__main__':
    with open("inputs/input1.txt", "r") as f:
        arr = list(map(lambda x: (x.split()[0], int(x.split()[1])), f.read().splitlines()))
        hor, depth, aim = 0, 0, 0
        for dir, magnitude in arr:
            hor += magnitude if dir == 'forward' else 0
            depth += (magnitude*aim) if dir == 'forward' else 0
            aim += magnitude *(1 if dir == 'down' else -1 if dir=='up' else 0)
        print(hor*depth)
